src/Microsoft.AspNet.JsonPatch/Adapters/ObjectAdapter.cs

```diff
@@ -81,16 +81,21 @@ public ObjectAdapter(IContractResolver contractResolver)
         /// </summary>
         /// <param name="operation">The add operation</param>
         /// <param name="objectApplyTo">Object to apply the operation to</param>
-        public void Add(Operation<T> operation, T objectToApplyTo)
+        public void Add(Operation<T> operation, T objectToApplyTo, Action<string> action = null)
```

Comment:
The error reporting action could be a member of the `ObjectAdapter` instance, rather than a part of the interface.